src/pynwb/file.py

```diff
@@ -87,8 +95,10 @@ def __init__(self, **kwargs):
                        "date_of_birth",
                        "strain")
         args_to_set = popargs_to_dict(keys_to_set, kwargs)
-        kwargs['name'] = 'subject'
-        super().__init__(**kwargs)
+        super().__init__(name="subject", **kwargs)
+
+        if args_to_set["age__reference"] not in (None, "birth", "gestational"):
```

Comment:
For fields like this where we have an explicit limited list of allowable values, I think it could be useful to specify that list as a class variable or class method to get the list of allowable values. In this way users can more easily inspect which values are permitted.

Reply:
I'm not 100% sure, but I don't think the case ``args_to_set["age__reference"]  == None`` should ever occur. If ``age__reference`` is missing in the file, then the fact that ``birth`` is the default value means that you should see ``birth`` as the value in this case. As such, the only time it would be ``None`` is if the user explicitly sets the value to ``None``. Is this a case we should allow?

Reply:
@oruebel 

default already has the behavior you are looking for
```python
from datetime import datetime
from dateutil.tz import tzlocal, tzutc

from pynwb.file import NWBFile, Subject

from pynwb import NWBHDF5IO

subject = Subject(
    age='P90D',
    #age__reference="birth",
    subject_id='RAT123',
)
nwbfile = NWBFile(
    'a test session description for a test NWBFile',
    'FILE123',
    datetime(2017, 5, 1, 12, tzinfo=tzlocal()),
    session_id='test1',
    subject=subject,
)

with NWBHDF5IO("test_age_ref.nwb", "w") as io:
    io.write(nwbfile)
    
with NWBHDF5IO("test_age_ref.nwb", "r") as io:
    nwb = io.read()
    print(nwb.subject.age__reference)
```

```python
None
```